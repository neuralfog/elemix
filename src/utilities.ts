export const ref = <Value>(
    value: Value | undefined = undefined,
): { value?: Value } => {
    return {
        value,
    };
};
