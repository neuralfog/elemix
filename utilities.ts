export { mergeClasses } from '@neuralfog/elemix-renderer/utils';
export * from './src/utilities';
