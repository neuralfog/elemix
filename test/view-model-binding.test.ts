import { expect, test, describe, beforeEach } from 'vitest';
import { HTML } from '@neuralfog/elemix-testing/snapshots';
import { html } from '@neuralfog/elemix-renderer';
import { Present } from '@neuralfog/elemix-testing';
import type { ViewModelBinding } from './fixtures/ViewModelBinding';
import { render } from '../utilities';

import './fixtures/ViewModelBinding';

describe('View Model Binding', () => {
    beforeEach(() => {
        document.body.innerHTML = '';
    });

    test('Initial State', async () => {
        const presenter = new Present().screen(
            html`<view-model-binding-app />`,
        );
        const viewModelApp = presenter.root<ViewModelBinding>();

        await render();

        expect(HTML(viewModelApp)).toMatchSnapshot();
    });

    test('Update Input Value', async () => {
        const presenter = new Present().screen(
            html`<view-model-binding-app />`,
        );
        const viewModelApp = presenter.root<ViewModelBinding>();

        await render();

        const input = presenter.getByTag<HTMLInputElement>('input');
        input.value = 'Brown Hounds';

        const event = new Event('input', {
            bubbles: true,
        });
        input.dispatchEvent(event);

        await render();

        expect(viewModelApp.state.input.value).toBe('Brown Hounds');
        expect(HTML(viewModelApp)).toMatchSnapshot();
    });

    test('Update View Model', async () => {
        const presenter = new Present().screen(
            html`<view-model-binding-app />`,
        );
        const viewModelApp = presenter.root<ViewModelBinding>();
        viewModelApp.state.input.value = 'Brown Hounds';

        await render();

        expect(HTML(viewModelApp)).toMatchSnapshot();

        const input = presenter.getByTag<HTMLInputElement>('input');
        expect(input.value).toBe('Brown Hounds');
    });
});
