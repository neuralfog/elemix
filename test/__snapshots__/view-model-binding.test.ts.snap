// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`View Model Binding > Initial State 1`] = `
"
<view-model-binding-app><!-- #shadowroot -->
  <h1>RefApp</h1>
  <p></p><input type="text">
</view-model-binding-app>
"
`;

exports[`View Model Binding > Update Input Value 1`] = `
"
<view-model-binding-app><!-- #shadowroot -->
  <h1>RefApp</h1>
  <p>Brown Hounds</p><input type="text">
</view-model-binding-app>
"
`;

exports[`View Model Binding > Update View Model 1`] = `
"
<view-model-binding-app><!-- #shadowroot -->
  <h1>RefApp</h1>
  <p>Brown Hounds</p><input type="text">
</view-model-binding-app>
"
`;
